cachi2/core/package_managers/pip.py

```diff
@@ -1800,17 +1827,16 @@ def _process_package_distributions(
             user_checksums,
         )
 
-        if dpi.package_type == "sdist":
-            sdists.append(dpi)
+        if dpi.should_download():
+            artifacts.append(dpi)
         else:
-            if dpi.should_download_wheel():
-                wheels.append(dpi)
-            else:
-                log.info("Filtering out %s due to checksum mismatch", package.filename)
+            log.info("Filtering out %s due to checksum mismatch", package.filename)
 
-    if len(sdists) != 0:
-        best_sdist = max(sdists, key=_sdist_preference)
-        if best_sdist.is_yanked:
+    sdists = [dpi for dpi in artifacts if dpi.package_type == "sdist"]
+    wheels = [dpi for dpi in artifacts if dpi.package_type == "wheel"]
```

Comment:
This split of artifacts could be done in the previous for loop. Something like:

```
if dpi.should_download():
    if dpi.package == "sdist":
        sdists.append(dpi)
   elif ...
```
As a result, we will not need the `artifacts` variable anymore.